odoo/addons/base/models/res_users.py

```diff
@@ -1450,36 +1450,28 @@ def fields_get(self, allfields=None, attributes=None):
 
                 # FIXME: in Accounting, the groups in the selection are not
                 # totally ordered, and their order therefore partially depends
-                # on their name, which is translated!  So we generate all the
-                # possible field names according to the partial order.
-                gs_list = [gs]
+                # on their name, which is translated!  However, the group name
+                # used in the "user groups view" corresponds to the group order
+                # without translations.
+                field_name_gs = gs
                 if app.xml_id == 'base.module_category_accounting_accounting':
-                    # ranks = {0: [A, B], 2: [C], 3: [D]}
-                    ranks = defaultdict(list)
-                    for g in gs:
-                        ranks[len(g.trans_implied_ids & gs)].append(g)
-                    # perms = [[AB, BA], [C], [D]]
-                    perms = [
-                        [Group.concat(*perm) for perm in itertools.permutations(rank)]
-                        for k, rank in sorted(ranks.items())
-                    ]
-                    # gs_list = [ABCD, BACD]
-                    gs_list = [Group.concat(*perm) for perm in itertools.product(*perms)]
-
-                for gs in gs_list:
-                    field_name = name_selection_groups(gs.ids)
-                    if allfields and field_name not in allfields:
-                        continue
-                    # selection group field
-                    tips = ['%s: %s' % (g.name, g.comment) for g in gs if g.comment]
-                    res[field_name] = {
-                        'type': 'selection',
-                        'string': app.name or _('Other'),
-                        'selection': selection_vals + [(g.id, g.name) for g in gs],
-                        'help': '\n'.join(tips),
-                        'exportable': False,
-                        'selectable': False,
-                    }
+                    # put field_name_gs in the same order as in the user form view
+                    order = {g: len(g.trans_implied_ids & gs) for g in gs}
+                    field_name_gs = gs.with_context(lang=None).sorted('name').sorted(order.get)
```

Comment:
Shouldn't it be done for all "app"? I don't see why it needs to be specialized for `base.module_category_accounting_accounting` only.

Reply:
By design we use a selection field only when the corresponding groups are totally ordered.
However, an exception was introduced for the accounting groups: they are not totally ordered but we still use a selection field.  So this is only needed for that category of groups.